src/vs/workbench/contrib/externalTerminal/browser/externalTerminal.contribution.ts

```diff
@@ -26,114 +26,195 @@ import { LifecyclePhase } from 'vs/workbench/services/lifecycle/common/lifecycle
+							integratedTerminalService.setActiveInstance(instance);
+							terminalGroupService.showPanel(true);
+						}
 					}
+				} else if (externalTerminalService) {
+					distinct(targets.map(({ stat }) => stat!.isDirectory ? stat!.resource.fsPath : dirname(stat!.resource.fsPath))).forEach(cwd => {
+						externalTerminalService!.openTerminal(config.terminal.external, cwd);
+					});
 				}
-			} else if (externalTerminalService) {
-				distinct(targets.map(({ stat }) => stat!.isDirectory ? stat!.resource.fsPath : dirname(stat!.resource.fsPath))).forEach(cwd => {
-					externalTerminalService!.openTerminal(config.terminal.external, cwd);
-				});
-			}
-		});
-	}
-});
+			});
+		}
+	});
+}
+
+registerOpenTerminalCommand(OPEN_IN_INTEGRATED_TERMINAL_COMMAND_ID, 'integrated');
+registerOpenTerminalCommand(OPEN_IN_EXTERNAL_TERMINAL_COMMAND_ID, 'external');
 
 export class ExternalTerminalContribution extends Disposable implements IWorkbenchContribution {
-	private _openInTerminalMenuItem: IMenuItem;
+	private _openInIntegratedTerminalMenuItem: IMenuItem;
+	private _openInExternalTerminalMenuItem: IMenuItem;
+
+	private _IntegratedOpenEditors?: IDisposable;
+	private _IntegratedExplorer?: IDisposable;
+
+	private _ExternalOpenEditors?: IDisposable;
+	private _ExternalExplorer?: IDisposable;
 
 	constructor(
 		@IConfigurationService private readonly _configurationService: IConfigurationService
 	) {
 		super();
 
-		this._openInTerminalMenuItem = {
+		this._openInIntegratedTerminalMenuItem = {
 			group: 'navigation',
 			order: 30,
 			command: {
-				id: OPEN_IN_TERMINAL_COMMAND_ID,
+				id: OPEN_IN_INTEGRATED_TERMINAL_COMMAND_ID,
 				title: nls.localize('scopedConsoleAction', "Open in Terminal")
 			},
 			when: ContextKeyExpr.or(ResourceContextKey.Scheme.isEqualTo(Schemas.file), ResourceContextKey.Scheme.isEqualTo(Schemas.vscodeRemote))
 		};
-		MenuRegistry.appendMenuItem(MenuId.OpenEditorsContext, this._openInTerminalMenuItem);
-		MenuRegistry.appendMenuItem(MenuId.ExplorerContext, this._openInTerminalMenuItem);
+
+
+		this._openInExternalTerminalMenuItem = {
+			group: 'navigation',
+			order: 31,
+			command: {
+				id: OPEN_IN_EXTERNAL_TERMINAL_COMMAND_ID,
+				title: nls.localize('scopedConsoleAction.external', "Open in External Terminal")
+			},
+			when: ContextKeyExpr.or(ResourceContextKey.Scheme.isEqualTo(Schemas.file), ResourceContextKey.Scheme.isEqualTo(Schemas.vscodeRemote))
+		};
+
 
 		this._configurationService.onDidChangeConfiguration(e => {
 			if (e.affectsConfiguration('terminal.explorerKind') || e.affectsConfiguration('terminal.external')) {
 				this._refreshOpenInTerminalMenuItemTitle();
+				this._refreshOpenInTerminalMenuItems();
 			}
 		});
+
 		this._refreshOpenInTerminalMenuItemTitle();
+		this._refreshOpenInTerminalMenuItems();
+	}
+
+	private _refreshOpenInTerminalMenuItems(): void {
```

Comment:
We should be able to use the `when` in the menu item to check if the setting equals a particular value. When we do this we could also check if it's a remote workspace and make sure only a single one shows there as external terminals aren't supported in that case (you can run the new testresolver window command to test in remote)

Reply:
oh! It's much more convenient this way